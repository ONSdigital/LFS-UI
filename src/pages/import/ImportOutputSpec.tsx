import React, {Component} from "react";
import DocumentTitle from "react-document-title";
import {ImportOutputSpecTable} from "./ImportOutputSpecTable";
import {ONSBreadcrumbs} from "../../components/ONS_DesignSystem/ONSBreadcrumbs"

interface Props {
    match: any
}

export class ImportOutputSpec extends Component <Props, {}> {
    displayName = ImportOutputSpec.name;

    render() {
        return (
            <DocumentTitle title={"LFS Imports Overview"}>
                <div className="container">
                    <ONSBreadcrumbs List={[{name: "Import Overview", link: "import/overview"}]} Current="Output File Specification"></ONSBreadcrumbs>
                    <h3>Output File Specification</h3>
                    <ImportOutputSpecTable/>
                </div>
            </DocumentTitle>
        );
    }
}