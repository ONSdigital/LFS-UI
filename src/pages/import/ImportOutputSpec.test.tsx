import React from "react";
import {cleanup, render} from "@testing-library/react";

import {BrowserRouter} from "react-router-dom";

import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import flushPromises from "../../tests/util/flushPromises";
import fetch from "../../tests/setup/__mocks__/fetch";
import {ImportOutputSpec} from "./ImportOutputSpec";
import MockDate from "mockdate";

// @ts-ignore
global.fetch = fetch;
MockDate.set('2015-01-21');


describe("Import Output Specification Page", () => {
    Enzyme.configure({adapter: new Adapter()});

    beforeEach(() => {
        // @ts-ignore
        process.env.NODE_ENV = "test";
    });

    afterEach(() => {
        cleanup();
    });


    function wrapper(render: any) {
        return render(
            <BrowserRouter>
                <ImportOutputSpec match={""}/>
            </BrowserRouter>
        );
    }

    it("renders the page", async () => {
        expect(wrapper(render)).toMatchSnapshot();
    });

    it("should render without crashing", async () => {
        const {queryAllByTestId, getByText} = wrapper(render);
    });

    it("should display the page name in the Breadcrumbs", async () => {
        const {getByTestId} = wrapper(render);

        await flushPromises();

        let currentBreadcrumb = getByTestId(/breadcrumb-current/i);

        expect(currentBreadcrumb.textContent).toEqual("Output File Specification");
    });

    it("should display the list of all imports", async () => {
        const {getByTestId, queryAllByTestId, getByText} = wrapper(render);

        await flushPromises();

        let list = queryAllByTestId(/table-row/i);

        expect(list.length).toEqual(5);

        expect(getByText(/APS Person Variable Specification/i)).toBeDefined();
        expect(getByText(/APS Household Variable Specification/i)).toBeDefined();
        expect(getByText(/Eurostat Variable Specification/i)).toBeDefined();
        expect(getByText(/LFS Household Variable Specification/i)).toBeDefined();
        expect(getByText(/LFS Person Variable Specification/i)).toBeDefined();
    });
});