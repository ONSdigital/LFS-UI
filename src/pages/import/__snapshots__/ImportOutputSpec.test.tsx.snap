// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Import Output Specification Page renders the page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <nav
          aria-label="Breadcrumb"
          class="breadcrumb"
          data-testid="breadcrumbs"
        >
          <ol
            class="breadcrumb__items"
          >
            <li
              class="breadcrumb__item u-fs-s"
              id="breadcrumb-1"
            >
              <a
                class="breadcrumb__link"
                href="/import/overview"
              >
                Import Overview
              </a>
            </li>
            <li
              class="breadcrumb__item u-fs-s breadcrumb__item--current"
              data-testid="breadcrumb-current"
              id="breadcrumb-2"
            >
              Output File Specification
            </li>
          </ol>
        </nav>
        <h3>
          Output File Specification
        </h3>
        <div>
          <table
            class="table table--sortable"
            data-aria-asc="ascending"
            data-aria-desc="descending"
            data-aria-sort="Sort by"
            id="basic-table"
          >
            <thead
              class="table__head"
            >
              <tr
                class="table__row"
              >
                <th
                  aria-sort="none"
                  class="table__header "
                  scope="col"
                >
                  <button
                    aria-label="Sort by File Name"
                    class="table__sort-button"
                    data-index="0"
                    type="button"
                  >
                    File Name
                  </button>
                </th>
                <th
                  aria-sort="none"
                  class="table__header "
                  scope="col"
                >
                  <button
                    aria-label="Sort by Import Date"
                    class="table__sort-button"
                    data-index="1"
                    type="button"
                  >
                    Import Date
                  </button>
                </th>
                <th
                  aria-sort="none"
                  class="table__header "
                  scope="col"
                >
                  <button
                    aria-label="Sort by Status"
                    class="table__sort-button"
                    data-index="2"
                    type="button"
                  >
                    Status
                  </button>
                </th>
                <th
                  class="table__header "
                  scope="col"
                />
              </tr>
            </thead>
            <tbody
              class="table__body "
            >
              <tr
                class="table__row nonSelectableTableRow"
                data-testid="table-row"
                tabindex="0"
                title="Table Row"
              >
                <td
                  class="table__cell "
                >
                  APS Person Variable Specification
                </td>
                <td
                  class="table__cell "
                >
                  18/06/1955
                </td>
                <td
                  class="table__cell "
                >
                  <span
                    class="status status--info"
                  >
                    Imported
                  </span>
                </td>
                <td
                  class="table__cell "
                >
                  <a
                    class="breadcrumb__link"
                    href="/import/APS Person Variable Specification"
                  >
                    Import
                  </a>
                </td>
              </tr>
              <tr
                class="table__row nonSelectableTableRow"
                data-testid="table-row"
                tabindex="0"
                title="Table Row"
              >
                <td
                  class="table__cell "
                >
                  APS Household Variable Specification
                </td>
                <td
                  class="table__cell "
                >
                  18/06/1955
                </td>
                <td
                  class="table__cell "
                >
                  <span
                    class="status status--info"
                  >
                    Imported
                  </span>
                </td>
                <td
                  class="table__cell "
                >
                  <a
                    class="breadcrumb__link"
                    href="/import/APS Household Variable Specification"
                  >
                    Import
                  </a>
                </td>
              </tr>
              <tr
                class="table__row nonSelectableTableRow"
                data-testid="table-row"
                tabindex="0"
                title="Table Row"
              >
                <td
                  class="table__cell "
                >
                  Eurostat Variable Specification
                </td>
                <td
                  class="table__cell "
                >
                  18/06/1955
                </td>
                <td
                  class="table__cell "
                >
                  <span
                    class="status status--info"
                  >
                    Not Imported
                  </span>
                </td>
                <td
                  class="table__cell "
                >
                  <a
                    class="breadcrumb__link"
                    href="/import/Eurostat Variable Specification"
                  >
                    Import
                  </a>
                </td>
              </tr>
              <tr
                class="table__row nonSelectableTableRow"
                data-testid="table-row"
                tabindex="0"
                title="Table Row"
              >
                <td
                  class="table__cell "
                >
                  LFS Household Variable Specification
                </td>
                <td
                  class="table__cell "
                >
                  18/06/1955
                </td>
                <td
                  class="table__cell "
                >
                  <span
                    class="status status--info"
                  >
                    File Older than One year
                  </span>
                </td>
                <td
                  class="table__cell "
                >
                  <a
                    class="breadcrumb__link"
                    href="/import/LFS Household Variable Specification"
                  >
                    Import
                  </a>
                </td>
              </tr>
              <tr
                class="table__row nonSelectableTableRow"
                data-testid="table-row"
                tabindex="0"
                title="Table Row"
              >
                <td
                  class="table__cell "
                >
                  LFS Person Variable Specification
                </td>
                <td
                  class="table__cell "
                >
                  18/06/1955
                </td>
                <td
                  class="table__cell "
                >
                  <span
                    class="status status--info"
                  >
                    Imported
                  </span>
                </td>
                <td
                  class="table__cell "
                >
                  <a
                    class="breadcrumb__link"
                    href="/import/LFS Person Variable Specification"
                  >
                    Import
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <nav
        aria-label="Breadcrumb"
        class="breadcrumb"
        data-testid="breadcrumbs"
      >
        <ol
          class="breadcrumb__items"
        >
          <li
            class="breadcrumb__item u-fs-s"
            id="breadcrumb-1"
          >
            <a
              class="breadcrumb__link"
              href="/import/overview"
            >
              Import Overview
            </a>
          </li>
          <li
            class="breadcrumb__item u-fs-s breadcrumb__item--current"
            data-testid="breadcrumb-current"
            id="breadcrumb-2"
          >
            Output File Specification
          </li>
        </ol>
      </nav>
      <h3>
        Output File Specification
      </h3>
      <div>
        <table
          class="table table--sortable"
          data-aria-asc="ascending"
          data-aria-desc="descending"
          data-aria-sort="Sort by"
          id="basic-table"
        >
          <thead
            class="table__head"
          >
            <tr
              class="table__row"
            >
              <th
                aria-sort="none"
                class="table__header "
                scope="col"
              >
                <button
                  aria-label="Sort by File Name"
                  class="table__sort-button"
                  data-index="0"
                  type="button"
                >
                  File Name
                </button>
              </th>
              <th
                aria-sort="none"
                class="table__header "
                scope="col"
              >
                <button
                  aria-label="Sort by Import Date"
                  class="table__sort-button"
                  data-index="1"
                  type="button"
                >
                  Import Date
                </button>
              </th>
              <th
                aria-sort="none"
                class="table__header "
                scope="col"
              >
                <button
                  aria-label="Sort by Status"
                  class="table__sort-button"
                  data-index="2"
                  type="button"
                >
                  Status
                </button>
              </th>
              <th
                class="table__header "
                scope="col"
              />
            </tr>
          </thead>
          <tbody
            class="table__body "
          >
            <tr
              class="table__row nonSelectableTableRow"
              data-testid="table-row"
              tabindex="0"
              title="Table Row"
            >
              <td
                class="table__cell "
              >
                APS Person Variable Specification
              </td>
              <td
                class="table__cell "
              >
                18/06/1955
              </td>
              <td
                class="table__cell "
              >
                <span
                  class="status status--info"
                >
                  Imported
                </span>
              </td>
              <td
                class="table__cell "
              >
                <a
                  class="breadcrumb__link"
                  href="/import/APS Person Variable Specification"
                >
                  Import
                </a>
              </td>
            </tr>
            <tr
              class="table__row nonSelectableTableRow"
              data-testid="table-row"
              tabindex="0"
              title="Table Row"
            >
              <td
                class="table__cell "
              >
                APS Household Variable Specification
              </td>
              <td
                class="table__cell "
              >
                18/06/1955
              </td>
              <td
                class="table__cell "
              >
                <span
                  class="status status--info"
                >
                  Imported
                </span>
              </td>
              <td
                class="table__cell "
              >
                <a
                  class="breadcrumb__link"
                  href="/import/APS Household Variable Specification"
                >
                  Import
                </a>
              </td>
            </tr>
            <tr
              class="table__row nonSelectableTableRow"
              data-testid="table-row"
              tabindex="0"
              title="Table Row"
            >
              <td
                class="table__cell "
              >
                Eurostat Variable Specification
              </td>
              <td
                class="table__cell "
              >
                18/06/1955
              </td>
              <td
                class="table__cell "
              >
                <span
                  class="status status--info"
                >
                  Not Imported
                </span>
              </td>
              <td
                class="table__cell "
              >
                <a
                  class="breadcrumb__link"
                  href="/import/Eurostat Variable Specification"
                >
                  Import
                </a>
              </td>
            </tr>
            <tr
              class="table__row nonSelectableTableRow"
              data-testid="table-row"
              tabindex="0"
              title="Table Row"
            >
              <td
                class="table__cell "
              >
                LFS Household Variable Specification
              </td>
              <td
                class="table__cell "
              >
                18/06/1955
              </td>
              <td
                class="table__cell "
              >
                <span
                  class="status status--info"
                >
                  File Older than One year
                </span>
              </td>
              <td
                class="table__cell "
              >
                <a
                  class="breadcrumb__link"
                  href="/import/LFS Household Variable Specification"
                >
                  Import
                </a>
              </td>
            </tr>
            <tr
              class="table__row nonSelectableTableRow"
              data-testid="table-row"
              tabindex="0"
              title="Table Row"
            >
              <td
                class="table__cell "
              >
                LFS Person Variable Specification
              </td>
              <td
                class="table__cell "
              >
                18/06/1955
              </td>
              <td
                class="table__cell "
              >
                <span
                  class="status status--info"
                >
                  Imported
                </span>
              </td>
              <td
                class="table__cell "
              >
                <a
                  class="breadcrumb__link"
                  href="/import/LFS Person Variable Specification"
                >
                  Import
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
